### PR TITLE
[doc] Logstash Kubernetes - Securing Logstash  

### PR DESCRIPTION
**Pull request by kaisecheng**
Fixes: #14580 14580

- Using certificates
- Using Kubernetes secrets
- Using the logstash keystore
- Securing ES on ECK

### CODE REVIEW

File: docsk8s/setting-up/ls-k8s-secure.asciidoc [hunk 1]
```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
...
-[[security-set-up-tls]]
+[[security-tls]]
 ==== Setting up TLS
 
-It's important to set up transport layer security (TLS) to ensure safe communication between the Elastic Stack components running in Kubernetes.
+It's important to set up transport layer security (TLS) to ensure safe communication between the Elastic Stack components running in {k8s}.
+Let's take {filebeat} and {ls} TLS mutual verification as an example. {ls} serves as the server side, while {filebeat} is the client.
 
-[[security-eck-secrets]]
+Create a `Secret` containing server and client SSH keys:
```
Comment (robbavey) [inline docsk8s/setting-up/ls-k8s-secure.asciidoc:14]: Typo

```suggestion
Create a `Secret` containing server and client SSL keys:
```

File: docsk8s/setting-up/ls-k8s-secure.asciidoc [hunk 2]
```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
...
 
-==== Using secrets from an ECK installation
-Description.
+[source,sh]
+--
+kubectl create secret generic logstash-beats-tls --from-file=ca.crt --from-file=client.crt --from-file=client.key --from-file=server.crt --from-file=server.pkcs8.key
+--
 
-[[security-logstash-keystore]]
+On {ls}, configurate the server certificates to the pipeline:
```
Comment (robbavey) [inline docsk8s/setting-up/ls-k8s-secure.asciidoc:21]:
```suggestion
On {ls}, configure the server certificates to the pipeline:
```

File: docsk8s/setting-up/ls-k8s-secure.asciidoc [hunk 3]
```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
...
+        volumes:
+          - name: logstash-beats-tls
+            secret:
+              secretName: logstash-beats-tls
+--
+
+[[security-k8s-secret]]
+==== Using secret
+
+One way to config confidential data, such as password, is to use `Secret` and references it as container environment variable.
```
Comment (robbavey) [inline docsk8s/setting-up/ls-k8s-secure.asciidoc:98]: We should probably add a note here that this is for illustration purposes and, in production, managing k8s secrets should be done using recognized [good practices](https://kubernetes.io/docs/concepts/security/secrets-good-practices/) to ensure the protection of sensitive information.

File: docsk8s/setting-up/ls-k8s-secure.asciidoc [hunk 4]
```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
...
+            - name: logstash-keystore
+              mountPath: /usr/share/logstash/config/logstash.keystore
+              subPath: logstash.keystore
+      volumes:
+        - name: logstash-keystore
+          secret:
+            secretName: logstash-keystore
+--
+
+<1> `LOGSTASH_KEYSTORE_PASS` is required only when the keystore is protected by {logstash-ref}/keystore.html#keystore-password[password]
```
Comment (robbavey) [inline docsk8s/setting-up/ls-k8s-secure.asciidoc:173]: Maybe remove "only", as we recommend the use of password protection in production

```suggestion
<1> `LOGSTASH_KEYSTORE_PASS` is required when the keystore is protected by {logstash-ref}/keystore.html#keystore-password[password]
```

File: docsk8s/setting-up/ls-k8s-secure.asciidoc [hunk 5]
```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
 
 In order to prepare your environment to be production ready, you'll need to set up secure communication between each of your Elastic resources.
 
-[[security-set-up-tls]]
+[[security-tls]]
 ==== Setting up TLS
 
-It's important to set up transport layer security (TLS) to ensure safe communication between the Elastic Stack components running in Kubernetes.
+It's important to set up transport layer security (TLS) to ensure safe communication between the Elastic Stack components running in {k8s}.
+Let's take {filebeat} and {ls} TLS mutual verification as an example. {ls} serves as the server side, while {filebeat} is the client.
```
Comment (robbavey) [inline docsk8s/setting-up/ls-k8s-secure.asciidoc:12]: I wonder if we should either link to the [securing communication between beats and logstash page](https://www.elastic.co/guide/en/beats/filebeat/current/configuring-ssl-logstash.html) or have the same caveat that "Creating a correct SSL/TLS infrastructure is outside the scope of this document" that that document does

File: docsk8s/setting-up/ls-k8s-secure.asciidoc [hunk 6]
```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
...
+            secret:
+              secretName: logstash-beats-tls
+--
+
+[[security-k8s-secret]]
+==== Using secret
+
+One way to config confidential data, such as password, is to use `Secret` and references it as container environment variable.
 
+Encode confidential data with Base64:
```
Comment (robbavey) [inline docsk8s/setting-up/ls-k8s-secure.asciidoc:99]: Add note that `Base64` is encoding, not encryption?

File: docsk8s/setting-up/ls-k8s-secure.asciidoc [hunk 7]
```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
...
+<1> `LOGSTASH_KEYSTORE_PASS` is required only when the keystore is protected by {logstash-ref}/keystore.html#keystore-password[password]
+
+[[security-eck-secrets]]
+==== Securing connection to {es} on ECK
+
+
+[[security-eck-secrets-pw]]
+===== Authentication
+
+ECK create a user for each Elastic resources. To access the resources, such as {es}, {ls} needs username and password.
```
Comment (robbavey) [inline docsk8s/setting-up/ls-k8s-secure.asciidoc:182]:
```suggestion
ECK creates a user for every Elastic resource. To access these resources, such as {es}, {ls} needs a username and password. 
```

File: docsk8s/setting-up/ls-k8s-secure.asciidoc [hunk 8]
```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
...
+        volumes:
+          - name: logstash-beats-tls
+            secret:
+              secretName: logstash-beats-tls
+--
+
+[[security-k8s-secret]]
+==== Using secret
+
+One way to config confidential data, such as password, is to use `Secret` and references it as container environment variable.
```
Comment (robbavey) [inline docsk8s/setting-up/ls-k8s-secure.asciidoc:97]:
```suggestion
To store sensitive information, such as a password, we can use a {k8s} `Secret`, and reference it as a container environment variable.
```

**Review by robbavey — CHANGES_REQUESTED** [review]

File: docsk8s/setting-up/ls-k8s-secure.asciidoc [hunk 9]
```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
 
 In order to prepare your environment to be production ready, you'll need to set up secure communication between each of your Elastic resources.
 
-[[security-set-up-tls]]
+[[security-tls]]
 ==== Setting up TLS
 
-It's important to set up transport layer security (TLS) to ensure safe communication between the Elastic Stack components running in Kubernetes.
+It's important to set up transport layer security (TLS) to ensure safe communication between the Elastic Stack components running in {k8s}.
```
Comment (karenzone) [inline docsk8s/setting-up/ls-k8s-secure.asciidoc:11]:
```suggestion
Transport layer security (TLS) helps ensure safe communication between the {stack} components running in {k8s}.
```

File: docsk8s/setting-up/ls-k8s-secure.asciidoc (hunk 5 above)
Comment (kaisecheng) [reply docsk8s/setting-up/ls-k8s-secure.asciidoc:12]: 👍 link to filebeat doc

**Review by robbavey — APPROVED** [review]
LGTM - over to @karenzone

**Review by karenzone — APPROVED** [review]
Thanks for your awesome work on this! Let's revisit the stacked heading in a future revision. But for now, LGTM 🚀

**Pull request merged**